### ISSUE TITLE
windows 11, nvda does not announce the progress of updates in windows update 

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#12831]
### Steps to reproduce:
on windows 11 open settings and windows update.
If there are no updates look for new updates.
If there are new updates, start the update download and installation process. 
### Actual behavior:
nvda only announces download and then installation. 
### Expected behavior:
nvda should do something similar to windows 10, announcing the progress of each make, download and installation informing the progress bar values ​​like 10%, 20%, etc.
In fact, the ideal experience is that of old windows like 7, where nvda only announces the progress bar, in windows 10 the experience was never ideal, because nvda announces the name of the update and then the percentage.
But this in windows 11 in my view was worse, because now nvda only announces what happens, download and installation, but it does not provide the value of the progress bar, and the user is not sure if the download or installation is close to The end.
I would also like to suggest, if the nvda user selects the announcement option of the progress bars in the background, nvda announces the value of the progress bar even if the user is in another window, a behavior I expected that does not occur even in windows 10.
The announcement of the progress bar in the background is useful when the windows user sets the option to restart as soon as possible, when this option is checked, windows restarts right after the update is finished, and the nvda user does not know if this moment is near.
Several requests, but the most important and focus of this request is to give windows 11 the same experience as windows 10, where the user is informed of the progress of updates.
The other requests I ask if possible.
thanks 
#### NVDA installed/portable/running from source:
instaled
#### NVDA version:
nvda.exe, NVDA alpha-23729,f8f62b04

#### Windows version:
11 10.0.22000.176
#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no
#### If add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by josephsl** [comment]
Hi, it was possible to announce Windows Update progress bar in Windows 10 because it used a live region for it. I suspect this behavior is the same with Narrator – if Narrator too cannot announce update progress in Windows 11 (so far, this is the case), then this is something Microsoft should fix from their side. Thanks.

**Comment by Brian1Gaff** [comment]
Interesting you say that as Windows 10 when updating also did not announce 
ofr bleep during the last windows update either. This was a Toshiba Laptop 
which had had a factory default reinstall and then  each day or so I went in 
and looked for updates, all was working till the latest one. It did the 
update, but not the progress indicator. This might be worth looking at. 
Obviously strange things can happen in Microsoft updates for some and not 
others as we are all too aware.


***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.

**Comment by feerrenrut** [comment]
We view this as a minor issue, potentially best fixed by Microsoft (updating Windows setting UI). There has been similar issues in this dialog for past versions of Windows.

Removing the milestone and giving P3

**Comment by josephsl** [comment]
Hi,

May 2022 update: Windows Update live region for downloading updates is not implemented in Windows 11 Version 22H2 preview (as of build 22610). As soon as I hear otherwise (that is, Narrator begins announcing update status for individual items rather than a summary), I vote to suspend this issue (keep it open, or if requested, close it for now).

Thanks.

**Comment by fernando-jose-silva** [comment]
my vote would be for nv access to implement solutions along the lines of Implemented in Windows 10

**Comment by josephsl** [comment]
Hi, this requires collaboration with Microsoft and a way to bring some things from Windows App Essentials add-on. But since the live region really comes from Microsoft, I think the next best thing we can do is ask Redmond to do something, hopefully during coper development milestone. Thanks.

**Comment by fernando-jose-silva** [comment]
Yes, I understand and agree.
The only problem is that this solution would take more than 1 year to reach a user incider, if there is time we could give users a remedy to improve the pain when the ideal solution arrives.

**Comment by josephsl** [comment]
Hi,

It's known that Microsoft can backport things, so it may not take up to a year for the solution to be made available to everyone. At the moment Windows 11 does provide live region for summary text and the issue of system focus moving around is resolved in 22H2 preview.

Thanks.

**Comment by Adriani90** [comment]
@fernando-jose-silva are you still having this issue with NVDA 2024.4.1 on Windows 11 last update?

**Comment by fernando-jose-silva** [comment]
Yes, the problem still occurs.
Currently, nvda announces the number of updates that are being downloaded and installed, but does not read the percentage of progress of these tasks.
You can find this information using object navigation.

**Comment by josephsl** [comment]
Hi,

Actually, they are announced but you must set UIA event registration to "global" (English interface) as these are announced as name changes instead of live regions.

Thanks.

**Comment by josephsl** [comment]
Hi, it would work across languages. In order for update announcement to be heard, you must move system focus to update entries themevles. Thanks.

**Comment by fernando-jose-silva** [comment]
With the current nvda resources, is there a way to read these updates without the focus being on it?
If not, the suggestion would be to read these changes when the window is in focus.

**Comment by josephsl** [comment]
Hi, the suggestion proposed was the one implemented in Windows App Essentials but was removed due to complexity and unreliability. The alternative I can think of is persuading Microsoft to change the progress message to a progress bar control and raise value change notification/event, and that requires redesigning Windows Update settings interface. Thanks.

**Comment by fernando-jose-silva** [comment]
How sad, I think Microsoft will not do anything in this case.
If we can do something in nvda, perhaps it would be disabled by default, and enabled if the user wishes.
The result of an add-on.

**Comment by josephsl** [comment]
Hi,

June 2025 update: starting from a build in say, April, at least Versoi n24H2 has begun using live region change event ao announce update status changes. Is this what is requested?

Thanks.

**Comment by Adriani90** [comment]
This works properly in Windows 11 24 H2 last build or Windows 11 25 H2. Tested with NVDA 2025.3. Closing as works for me.
